lib/new-syntax/parser.lr

```diff
@@ -887,6 +887,10 @@ chain_expression_list : Ast.Expression[] = {
 projection_expression : Ast.Expression = {
     filter_expression;
 
+    '[' boolean_expression: or_filter ']' 'of' expr:filter_expression => {
```

Comment:
Btw, do we need to support multiple boolean questions at once?

Reply:
hmm, then we will need to somehow name the boolean questions since the order in natural language may not be the same as the order in code. 
we are going to have support for multiple queries, I'm tempted to keep this simple and use separate queries when we really need to...

Reply:
Yeah I agree that if we want compound queries at some point we might as well handle them as two separate queries with a good old semicolon.